projects/amd-loader/bin/util.js

```diff
@@ -13,18 +13,40 @@ const copy = (input, output) =>
 	});
 
 function run(binary, ...args) {
-	const {error, signal, status} = spawnSync(binary, args, {
+	return run.withEnv(null, binary, ...args);
+}
+
+run.withEnv = function (env, binary, ...args) {
+	const options = {
 		shell: true,
 		stdio: 'inherit',
-	});
+	};
+
+	if (env) {
+		options['env'] = {
+			...process.env,
+			...env,
```

Comment:
nit: should env override process.env? I would think it might be the other way around. Not sure what is typically done for that

Reply:
Yes, you are right: I'm breaking the usual expectations. 

The problem [here](https://github.com/liferay/liferay-frontend-projects/pull/1050/files#diff-03935a18504c29dc8a0c37cf7c460bdd348624e90ad919380a8640fb5d1f974fR8-R10) is that if someone sets `flavor` in the env, the build will build the same three times.

But, since this function (run) is generic, I'll make it work generically, then tweak the calling code :+1:

Reply:
If its too much of a headache, don't worry about it

Reply:
I've fixed it :rocket: